src/pilot/assign_role.py

```diff
@@ -1094,19 +1120,27 @@ def generate_osd_config_with_journals(controllers, osd_drives, ssds):
     return osd_config
 
 
-def get_pci_bus_number(spinner, controllers):
-    bus = None
+def get_by_path_device_name(physical_disk, controllers):
+    disk_index = physical_disk.description.split(" ")[1]
     for controller in controllers:
-        if controller.id in spinner.id:
-            bus = controller.bus.lower()
-            break
-    return bus
-
-
-def get_by_path_device_name(pci_bus_number, physical_disk):
-    return ('/dev/disk/by-path/pci-0000:'
-            '{pci_bus_number}:00.0-sas-0x{sas_address}-lun-0').format(
-                pci_bus_number=pci_bus_number,
+        if physical_disk.controller == controller.id and \
+                controller.model.startswith("PERC H740P"):
+            return ('/dev/disk/by-path/pci-0000:'
+                    '{pci_bus_number}:00.0-scsi-0:2:{disk_index}:0').format(
+                pci_bus_number=controller.bus.lower(),
+                disk_index=disk_index)
+        elif physical_disk.controller == controller.id and \
+                controller.model.startswith("PERC H730P"):
+            return ('/dev/disk/by-path/pci-0000:'
+                    '{pci_bus_number}:00.0-scsi-0:'
+                    '{channel}:{disk_index}:0').format(
+                pci_bus_number=controller.bus.lower(),
+                channel=2 if physical_disk == "online" else 0,
```

Comment:
In the line above, physical_disk is an object, so it will never be equal to "online".  Should this be physical_disk.raid_status?